### ISSUE TITLE
Refactor quest list to use bootstrap-table

### ISSUE THREAD

**Issue by tylerecouture** [bytedeck/bytedeck#1369]
Try to mimic existing quest list as close as possible, but utilizing [bootstrap-table](https://examples.bootstrap-table.com/index.html?bootstrap3) so that the list can be sorted, filtered, etc.

For some examples of standard bootstrap-table usage, see the Student list at profiles_manager/profile_list.html and the tag list at tags/list.html

The difficult part will be the accordion, i.e. collapsing each row.
- Here's a proof of concept I made a long time ago, that could be used for ideas: https://jsfiddle.net/mpr63ezm/40/
- Some more discussion here: https://github.com/wenzhixin/bootstrap-table/issues/964

To start, just added the default search (on name) and sorting (on all columns).  We'll be adding filters in a new issue.

I recommend creating a duplicate quest list view for this with it's own url we can access manually.  Once the feature is complete we can swap it out for the main quest list.

**Comment by Enprogames** [comment]
I just saw that you've assigned this to me. I will get started working on it right away.

**Comment by Enprogames** [comment]
I'm very new to JavaScript, but I think I understand the JSFiddle you provided. I created a new one that more closely mirrors the behaviour of the hackerspace website: https://jsfiddle.net/enprogames/cv8f2Lgh/10/

I've also spent quite a bit of time diving into the Django templates that currently exist for this, and I'm a bit confused about them. My current understanding of the `quest_manager` templates that relate to this functionality goes as follows:
- quests.html: Main page for displaying the quests page. Shows the "Available", "In Progress", "Completed", and "Draft" tabs depending on who is using the site, and uses each of the "tab_*" templates to display a list of quests depending on which tab is currently selected.
- quests2.html: A previous attempt at redoing the quests.html page? It doesn't seem to be using bootstrap-tables though.
- tab_quests_approvals.html: Displays quests awaiting approval, allowing staff or TAs to review and approve quests?
- tab_quests_available.html: Displays the quests that are currently available for a user to undertake.
- tab_quests_available2.html: An attempt at improving tab_quests_available.html?
- tab_quests_submission.html: Displays quests that are currently in progress or have been completed by the user?

It seems like each of these tabs do essentially the same thing, so it looks like to complete this issue, all of these will need to be refactored to use bootstrap-table? Also, maybe we should create an additional issue to consolidate these views in the future?

**Comment by tylerecouture** [comment]
To start, only worry about tab_quests_available.html.   You can reuse the existing `quest_list2` view which already has a url and uses templates: `quests2.html` (probably doesn't need to be modified) which includes `tab_quests_available2.html`.  

You should only need to modify the `tab_quests_available2.html` template, at least for a first go.  We can worry about all the other views later, this is the critical one.

There are some code comments at the top of `quest.html` and `quest_approval.html` which explain how the other sub-templates are used (for after we get tab_quests_available2.html refactored)

**Comment by tylerecouture** [comment]
A few minor remaining tweaks:

- [ ] Change Tag to Tags (plural) 
- [ ] Remove top line in all views (dark themes, and mobile view, doesn't appear in normal light theme)
![image](https://github.com/bytedeck/bytedeck/assets/10604391/172139ab-77f9-45a8-9d7d-aee01053ef7b)
- [ ] Small gap appearing (only for students?) at bottom of expanded accordians: 
![image](https://github.com/bytedeck/bytedeck/assets/10604391/dceb0d77-9cf0-468d-b6f5-4e7dffaf25fe)
